src/java.base/share/classes/sun/security/ssl/Hybrid.java

```diff
@@ -325,19 +316,34 @@ private record Handler(KEM.Encapsulator le, KEM.Encapsulator re,
         @Override
         public KEM.Encapsulated engineEncapsulate(int from, int to,
                 String algorithm) {
-            var left = le.encapsulate();
-            var right = re.encapsulate();
-            if (from == 0 && to == le.secretSize() + re.secretSize()) {
-                return new KEM.Encapsulated(
-                        new SecretKeyImpl(left.key(), right.key()),
-                        concat(left.encapsulation(), right.encapsulation()),
-                        null);
-            } else {
+            int expectedSecretSize = le.secretSize() + re.secretSize();
+            if (!(from == 0 && to == expectedSecretSize)) {
                 throw new IllegalArgumentException(
                         "Invalid range for encapsulation: from = " + from +
                         " to = " + to + ", expected total secret size = " +
-                        (le.secretSize() + re.secretSize()));
+                        expectedSecretSize);
             }
+
+            var left  = le.encapsulate();
+            var right = re.encapsulate();
+            byte[] leftEnc  = left.encapsulation();
+            byte[] rightEnc = right.encapsulation();
+
+            int expectedEncSize = le.encapsulationSize() +
+                    re.encapsulationSize();
+            int actualEncSize = leftEnc.length + rightEnc.length;
+
+            if (actualEncSize != expectedEncSize) {
```

Comment:
We probably don't need this check. `left` and `right` are generated inside this method and there is no need to worry about they have the wrong lengths.

This is not like in `engineDecapsulate` where `encapsulation` is provided by the caller and we need to validate it.

Reply:
Removed the extra len check.